extensions/wikia/CuratedContent/CuratedContentValidator.class.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+class CuratedContentValidator {
+	const LABEL_MAX_LENGTH = 48;
+
+	private $errors;
+	private $titles;
+	private $hadEmptyLabel;
+
+	public function __construct( $data ) {
+		$this->errors = [ ];
+		$this->titles = [ ];
+		$this->hadEmptyLabel = false;
+
+		// validate sections
+		foreach ( $data as $section ) {
+			if ( !empty( $section['featured'] ) ) {
+				$this->validateFeaturedSection( $section );
+			} else {
+				$this->validateSection( $section );
+			}
+		}
+		// also check section for duplicate title
+		foreach ( array_count_values( $this->titles ) as $title => $count ) {
+			if ( $count > 1 ) {
+				$this->error( [ 'title' => $title ], 'duplicatedLabel' );
+			}
+		}
+	}
+
+	private function error( $itemWithTitle, $errorString ) {
+		if ( array_key_exists( 'title', $itemWithTitle ) && !empty( $errorString ) ) {
+			$this->errors[ ] = [ 'title' => $itemWithTitle['title'], 'reason' => $errorString ];
+		}
+	}
+
+	public function getErrors() {
+		return $this->errors;
+	}
+
+	private function validateFeaturedSection( $section ) {
+		if ( !empty( $section['items'] ) && is_array( $section['items'] ) ) {
+			foreach ($section['items'] as $item) {
+				$this->validateItem($item);
+			}
+		}
+	}
+
+	private function validateImage( $sectionOrItem ) {
+		if ( empty( $sectionOrItem['image_id'] ) ) {
+			$this->error( $sectionOrItem, 'imageMissing' );
+		}
+	}
+
+	private function validateSection( $section ) {
+		// check for "optional" section - it has empty label, but there can be only ONE
+		if ( empty( $section['title'] ) ) {
+			if ( $this->hadEmptyLabel ) {
```

Comment:
maybe instead of calling this field hadEmptyLabel you should name it hasOptionalSection because it's only how you use this filed (to check if optional is set and if it's only one).